src/components/Checkbox/Checkbox.tsx

```diff
@@ -21,64 +18,59 @@ export class Checkbox extends React.Component<ICheckboxProps, ICheckboxState> {
   constructor(props: ICheckboxProps) {
     super(props);
 
-    this.state = {
-      isChecked: props.isChecked
-    };
-
-    this._id = `checkbox-${ _instance++ }`;
-
-    this._handleInputChange = this._handleInputChange.bind(this);
-  }
-
-  public componentWillReceiveProps(newProps: ICheckboxProps) {
-    if (newProps.isChecked !== this.props.isChecked) {
-      this.setState({
-        isChecked: newProps.isChecked
-      });
-    }
+    this._id = getId('checkbox-');
+    this._onChange = this._onChange.bind(this);
   }
 
   public render() {
-    let { text, isEnabled, className } = this.props;
-    let { isChecked } = this.state;
+    const {
+      checked,
+      className,
+      defaultChecked,
+      disabled,
+      inputProps,
+      label
+    } = this.props;
 
     return (
       <div className={ css('ms-ChoiceField', className) }>
         <input
-          ref={ (c): HTMLInputElement => this._checkBox = c }
+          { ...inputProps }
+          { ...(checked !== undefined && { checked }) }
+          { ...(defaultChecked !== undefined && { defaultChecked }) }
+          disabled={ disabled }
+          ref={ (el): HTMLInputElement => this._checkBox = el }
           id={ this._id }
           name={ this._id }
           className='ms-ChoiceField-input'
           type='checkbox'
           role='checkbox'
-          checked={ isChecked }
-          disabled={ !isEnabled }
-          onChange={ this._handleInputChange }
-          aria-checked={ isChecked }
+          onChange={ this._onChange }
+          aria-checked={ checked }
         />
         <label htmlFor={ this._id } className='ms-ChoiceField-field'>
-          { text && <span className='ms-Label'>{ text }</span> }
+          { label && <span className='ms-Label'>{ label }</span> }
         </label>
       </div>
     );
   }
 
+  public get checked(): boolean {
+    return this._checkBox ? this._checkBox.checked : false;
+  }
+
   public focus() {
       if (this._checkBox) {
           this._checkBox.focus();
       }
   }
 
-  private _handleInputChange(evt: React.FormEvent) {
-    let { onChanged } = this.props;
-    const isChecked = (evt.target as HTMLInputElement).checked;
-
-    this.setState({
-      isChecked: isChecked
-    });
+  private _onChange(ev: React.FormEvent) {
+    const { onChange } = this.props;
+    const isChecked = (ev.target as HTMLInputElement).checked;
```

Comment:
@dzearing when `checked = { true }` is sent as a Prop to Checkbox, isChecked here would always be 'false'. Resulting in callback to always return false. Vice versa.

As when we get the checked value from the Input element it would have have flipped to unchecked upon user interaction, though it will set back to checked but by then the callback would have returned with a 'false'.

In demo example
  `<Checkbox label='Selected item' onChange={ this._onCheckboxChange } checked={ true } />` 
logs 'The option has been changed to false.' upon interaction.